### ISSUE TITLE
Syntax enhancements to template API

### ISSUE THREAD

**Issue by mattsherman** [woothemes/woocommerce#39433]
Blocked by #39418, #39421, #39420

## Description

The `WooCommerceGroupableFormBlockTemplate` trait will support some syntax enhancements to allow for a more object-based usage.

## Implementation design

```php
trait WooCommerceGroupableFormBlockTemplate {
    public function get_section_by_id( string $id ): WooCommerceSection;
    public function get_section_by_name( string $name ): WooCommerceSection;

    public function get_group_by_id( string $id ): WooCommerceGroup; 
    public function get_group_by_name( string $name ): WooCommerceGroup;
}

class WooCommerceSection {
    public function add_field( array $field ): array;
}

class WooCommerceGroup {
   public function add_field( array $field ): array;
   public function add_section( array $section ): array;
}

// usage
$template
    ->get_group_by_id( 'GROUP_ID' )
    ->add_section( ... )
    ->add_field( ... );
```

## Acceptance criteria

- [ ] TODO

## Implementation notes

- PoC PR: https://github.com/woocommerce/woocommerce/pull/39129

**Comment by mattsherman** [comment]
Dropping this issue as it has been folded into other issues.
